### PR TITLE
Add a metric to track the number of managed files per column family in RocksDb

### PR DESCRIPTION
**Pull request by halfprice**
We don't add more detailed level to control the cardinality of the metrics.

Manually tested in local cluster:

Screenshot 2024-09-04 at 3 11 22 PM

## Description 

Describe the changes or additions included in this PR.

## Test plan 

How did you test the new or updated feature?

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:
- [ ] REST API:

### CODE REVIEW

File: crates/typed-store/src/rocks/mod.rs [hunk 1]
```diff
@@ -983,6 +984,17 @@ impl<K, V> DBMap<K, V> {
                 Self::get_int_property(rocksdb, &cf, ROCKSDB_PROPERTY_TOTAL_BLOB_FILES_SIZE)
                     .unwrap_or(METRICS_ERROR),
             );
+        let total_num_files: i64 = (0..=6)
+            .map(|level| {
```
Comment (sadhansood) [inline crates/typed-store/src/rocks/mod.rs:988]: Could we also query the num levels in the db or is this the max value?

**Review by sadhansood — APPROVED** [review]

File: crates/typed-store/src/rocks/mod.rs (hunk 1 above)
Comment (halfprice) [reply crates/typed-store/src/rocks/mod.rs:988]: This is the default value, and yes one can change it.

I didn't find a way to query db option in rust, but there is a c++ API that can do this.

**Review by mwtian — APPROVED** [review]
This should work but maybe we can parse the value from properties like `SSTABLES`. Also I think it should be ok cardinality-wise to have the level as a label, if only levels 0 to 6 are included.

File: crates/typed-store/src/rocks/mod.rs (hunk 1 above)
Comment (halfprice) [reply crates/typed-store/src/rocks/mod.rs:990]: Update comment about if we every change number of layers, we need to update here as well.

Worse case, it'll just miss lower level file count.

**Comment by halfprice** [comment]
> This should work but maybe we can parse the value from properties like `SSTABLES`. Alternatively I think it should be ok cardinality-wise to have the level as a label, if only levels 0 to 6 are included.

Hmm, didn't think of parsing from `SSTABLES` since there is a num_files helper property handy. Walrus shares the same DBMetrics, and cardinality is not ok there....

**Pull request merged**